vertx-pg-client/src/test/java/io/vertx/pgclient/junit/ContainerPgRule.java

```diff
@@ -0,0 +1,162 @@
+        .withUsername("postgres")
+        .withPassword("postgres")
+        .withCopyFileToContainer(MountableFile.forHostPath(setupFile.toPath()), "/docker-entrypoint-initdb.d/create-postgres.sql");
+  }
+  
+  private static File getTestResource(String name) throws Exception {
+    InputStream in = new FileInputStream(new File("docker" + File.separator + "postgres" + File.separator + "resources" + File.separator + name));
+    Path path = Files.createTempFile("pg-client", ".tmp");
+    Files.copy(in, path, StandardCopyOption.REPLACE_EXISTING);
+    File file = path.toFile();
+    file.deleteOnExit();
+    return file;
+  }
+
+  public static boolean isTestingWithExternalDatabase() {
+    return isSystemPropertyValid(connectionUri);
+  }
+
+  private static boolean isSystemPropertyValid(String systemProperty) {
+    return systemProperty != null && !systemProperty.isEmpty();
+  }
+
+  public synchronized PgConnectOptions startServer(String databaseVersion) throws Exception {
+    initServer(databaseVersion);
+    server.start();
+
+    return new PgConnectOptions()
+        .setPort(server.getMappedPort(PostgreSQLContainer.POSTGRESQL_PORT))
+        .setHost(server.getContainerIpAddress())
+        .setDatabase("postgres")
+        .setUser("postgres")
+        .setPassword("postgres");
+  }
+
+
+  private static String getPostgresVersion() {
+    String specifiedVersion = System.getProperty("embedded.postgres.version");
+    String version;
+    if (specifiedVersion == null || specifiedVersion.isEmpty()) {
+      // if version is not specified then V10.10 will be used by default
+      version = "10.10";
+    } else {
+      version = specifiedVersion;
+    }
+
+    return version;
+  }
+
+  public synchronized void stopServer() throws Exception {
+    if (server != null) {
+      try {
+        server.stop();
+      } finally {
+        server = null;
+      }
+    }
+  }
+
+  @Override
+  protected void before() throws Throwable {
+    // use an external database for testing
+    if (isTestingWithExternalDatabase()) {
+      options = PgConnectOptions.fromUri(connectionUri);
+      return;
+    }
+
+    // We do not need to launch another server if it's a shared instance
+    if (this.server != null) {
+      return;
+    }
+
+    this.databaseVersion =  getPostgresVersion();
+    options = startServer(databaseVersion);
+  }
+  
+  public boolean isAtLeastPg10() {
+    // hackish ;-)
+    return !getPostgresVersion().startsWith("9.");
+  }
+
+  @Override
+  protected void after() {
+    if (!isTestingWithExternalDatabase()) {
+      try {
+        stopServer();
+      } catch (Exception e) {
+        e.printStackTrace();
+      }
+    }
+  }
+
+  @Override
+  public Statement apply(Statement base, Description description) {
+    return new Statement() {
+      @Override
+      public void evaluate() throws Throwable {
+        if (!isAtLeastPg10()) {
+          throw new AssumptionViolatedException("postgresql >= 10 is needed ");
```

Comment:
can we use `Assume` in the testcase instead of checking the version here?

Reply:
I don't think so, the rule is loaded before the Assume in the test

Reply:
can you take https://github.com/eclipse-vertx/vertx-sql-client/blob/607e6ba04a250015be46d88917b8d11415bc3482/vertx-mysql-client/src/test/java/io/vertx/mysqlclient/MySQLUtilityCommandTest.java#L103 for reference?

Reply:
I have another PR (waiting for this one to be merged) which move the tests with embedded postgres to test container. 
Can this change be done later in the mentionned PR as it will require more change than just the `Assume` in the test.